src/qubx/trackers/riskctrl.py

```diff
@@ -95,37 +112,38 @@ def _get_price(update: float | Quote | Trade | Bar, direction: int) -> float:
     def update(
         self, ctx: StrategyContext, instrument: Instrument, update: Quote | Trade | Bar
     ) -> List[TargetPosition] | TargetPosition:
-        c = self._signals.get(instrument)
+        c = self._trackings.get(instrument)
         if c is None:
             return []
 
         match c.status:
-            case "NEW":
+            case State.NEW:
                 # - nothing to do just waiting for position to be open
                 pass
 
-            case "RISK-TRIGGERED":
+            case State.RISK_TRIGGERED:
                 # - nothing to do just waiting for position to be closed
                 pass
 
-            case "OPEN":
-                pos = ctx.positions[instrument.symbol].quantity
+            case State.OPEN:
+                pos = c.target.target_position_size
                 if c.signal.stop:
                     if (
                         pos > 0
                         and self._get_price(update, +1) <= c.signal.stop
                         or (pos < 0 and self._get_price(update, -1) >= c.signal.stop)
                     ):
-                        c.status = "RISK-TRIGGERED"
-                        logger.debug(f"\t ::: <red>Stop triggered</red> for {instrument.symbol} at {c.signal.stop}")
+                        c.status = State.RISK_TRIGGERED
+                        logger.debug(
+                            f"<yellow>{self.__class__.__name__}</yellow> triggered <red>STOP LOSS</red> for <green>{instrument.symbol}</green> at {c.signal.stop}"
+                        )
                         return TargetPosition.zero(
                             ctx,
                             instrument.signal(
                                 0,
                                 price=c.signal.stop,
                                 group="Risk Manager",
                                 comment="Stop triggered",
```

Comment:
this means that the basic stop taker tracker version will fill based on the market price right? and if we want exact fills that we need to use the advanced version.

Reply:
Yes, I want to test it bit more today on my strategy and then switch to AdvancedStopTakeTracker
One thing I didn't like here - that we use inheritance for AtrStopTake etc 
I think we need to use decorator pattern here and pass instance of basic logic as parameter

Reply:
true, that would be better